django/db/models/fields/computed.py

```diff
@@ -0,0 +1,131 @@
+from django.core import checks
+from django.db import connections
+from django.db.models.sql import Query
+
+from . import Field
+
+__all__ = ["ComputedField"]
+
+
+class ComputedField(Field):
+    computed = True
+    db_returning = True
+
+    _query = None
+    _resolved_expression = None
+    output_field = None
+
+    def __init__(self, expression, output_field=None, persisted=True, *args, **kwargs):
+        self.expression = expression
+        self._output_field = output_field
+        self.persisted = persisted
+
+        kwargs["editable"] = False
+        kwargs["blank"] = True
+        kwargs["default"] = None
+
+        super().__init__(*args, **kwargs)
+
+    def contribute_to_class(self, *args, **kwargs):
+        super().contribute_to_class(*args, **kwargs)
+
+        self._query = Query(model=self.model, alias_cols=False)
+        self._resolved_expression = self.expression.resolve_expression(
+            self._query, allow_joins=False
+        )
+        self.output_field = (
+            self._output_field
+            if self._output_field is not None
+            else self._resolved_expression.output_field
+        )
+
+    def _expression_sql(self, connection):
+        return self._resolved_expression.as_sql(
+            compiler=connection.ops.compiler("SQLCompiler")(
+                self._query, connection=connection, using=None
+            ),
+            connection=connection,
+        )
+
+    def check(self, **kwargs):
+        databases = kwargs.get("databases") or []
+        return [
+            *super().check(**kwargs),
+            *self._check_sql_expression(databases),
+            *self._check_persistence(databases),
+        ]
```

Comment:
IMO we could merge these into the `check()` function here, it would save a lot of temporary lists and looping over the databases list.

Reply:
I would tend to disagree here - every other `Field` in the django project follows this convention, so I recommend we do the same.

<img width="458" alt="Screen Shot 2023-03-25 at 6 23 50 PM" src="https://user-images.githubusercontent.com/3439947/227748606-354d09ca-b9b0-458b-a7c7-bd1f3e8b7cfd.png">
<img width="564" alt="Screen Shot 2023-03-25 at 6 23 21 PM" src="https://user-images.githubusercontent.com/3439947/227748595-3c2d4b44-740b-4639-9f41-01b5c94c748f.png">
<img width="623" alt="Screen Shot 2023-03-25 at 6 24 11 PM" src="https://user-images.githubusercontent.com/3439947/227748617-13538d62-b2cf-479d-8fcd-ba0ad7fef2d3.png">